runtime/src/iree-amd-aie/driver/xrt/direct_command_buffer.cc

```diff
@@ -348,8 +347,11 @@ static iree_status_t iree_hal_xrt_direct_command_buffer_dispatch(
         xrt::bo(*command_buffer->descriptor_sets[set].bindings[j],
                 command_buffer->descriptor_sets[set].lengths[j],
                 command_buffer->descriptor_sets[set].offsets[j]);
-    bos.push_back(arg_buffer);
     run.set_arg(arg_index + j, arg_buffer);
+    bool not_ofm = (bindings.values[j].buffer->memory_type & IREE_HAL_MEMORY_TYPE_HOST_VISIBLE) &&
+                   (bindings.values[j].buffer->allowed_usage & IREE_HAL_MEMORY_TYPE_HOST_VISIBLE);
```

Comment:
this is not how input/output is determined - https://github.com/nod-ai/iree-amd-aie/blob/makslevental/xrt-lite/runtime/src/iree-amd-aie/driver/xrt/cts/executable_cache_test.mlir#L21-L23

```mlir
      %0 = hal.interface.binding.subspan layout(#pipeline_layout) binding(0) alignment(64) offset(%c0) flags("ReadOnly|Indirect") : !flow.dispatch.tensor<readonly:tensor<32x32xf32>>
      %1 = hal.interface.binding.subspan layout(#pipeline_layout) binding(1) alignment(64) offset(%c0) flags("ReadOnly|Indirect") : !flow.dispatch.tensor<readonly:tensor<32x32xf32>>
      %2 = hal.interface.binding.subspan layout(#pipeline_layout) binding(2) alignment(64) offset(%c0) flags(Indirect) : !flow.dispatch.tensor<writeonly:tensor<32x32xf32>>
```

note `!flow.dispatch.tensor<readonly:tensor<32x32xf32>` vs `!flow.dispatch.tensor<writeonly:tensor<32x32xf32>`

Reply:
Also, as it clearly states, `IREE_HAL_MEMORY_TYPE_HOST_VISIBLE` is a `iree_hal_memory_type_bits_t` flag, not an `iree_hal_buffer_usage_bits_t` flag. See https://github.com/iree-org/iree/blob/fbf677d2096aea4c544cd8d0558b49bfc729fd91/runtime/src/iree/hal/buffer.h

Reply:
Please stop guessing - `IREE_HAL_BUFFER_USAGE_MAPPING` is not the correct flag: 

```
// WARNING: mapping can be extremely expensive, use limited hardware
  // resources, introduce data hazards, and synchronize host and device
  // execution. Unless an application knows that such issues will not arise
  // (as in tests where there's never concurrent usage) mapping should be used
  // judiciously: do not assume mapping is a high-performance technique!
```

https://github.com/iree-org/iree/blob/66342abbfaaee707e27ecc7d8151ad9e357ca0da/runtime/src/iree/hal/buffer.h#L389

Reply:
What I see is all input bo are created from here: 
![image](https://github.com/user-attachments/assets/b4de8471-0ada-41bc-aa54-6b73527a546f)


  mappable_params.type |= IREE_HAL_MEMORY_TYPE_HOST_VISIBLE;
  mappable_params.usage |= IREE_HAL_BUFFER_USAGE_MAPPING;

Reply:
Repeating again: you are looking at buffer_view APIs which is not the only place in HAL where buffers are created; we use `allovate_buffer` APIs:
 
https://github.com/nod-ai/iree-amd-aie/blob/main/runtime/src/iree-amd-aie/driver/xrt/direct_allocator.cc#L171

Reply:
Repeating again: **that is not the only place in HAL where bufferes are created**. 

Specifically those are the `buffer_view` APIs, which we do not use. We use the pure `allocate_buffer` APIs: https://github.com/nod-ai/iree-amd-aie/blob/fad96292e8fca71f588133bd914d1adf1d0b7953/runtime/src/iree-amd-aie/driver/xrt/direct_allocator.cc#L143

[[xrt_lite_dispatch_test.log](https://gist.github.com/makslevental/f1354802fad25c36722de75786f07e47)](https://gist.github.com/makslevental/f1354802fad25c36722de75786f07e47)

Reply:
Will this change affect xrt-lite? I thought this change only affect xrt

Reply:
![image](https://github.com/user-attachments/assets/110ff42d-f1af-4d47-b46f-6b1b76790215)

Reply:
Function iree_hal_buffer_view_generate_buffer_in_situ is at iree level. It will call xrt level function as you pointed out. I do not see a contradiction here.

Reply:
> It will call xrt level function as you pointed out. 

You're completely wrong. Feel free to run https://github.com/nod-ai/iree-amd-aie/blob/main/runtime/src/iree-amd-aie/driver/xrt/cts/executable_cache_test.cc and see for yourself.

Reply:
What I see is that the file you referenced has this line
_#include "iree/hal/buffer_view_util.h"_
That is aligned with my understanding.

Reply:
you're understanding is wrong https://github.com/nod-ai/iree-amd-aie/pull/852

feel free to consult any reference on how C++ headers work.

Reply:
Which test/application do you use to test _runtime/src/iree-amd-aie/driver/xrt/cts/matmul_dispatch_test.cc_?